paml/data.py

```diff
@@ -0,0 +1,66 @@
+"""
+Functions related to data i/o in connection with a protocol execution trace.
```

Comment:
Maybe add a comment at the top to point out that what you are doing here is not so much creating functions, as monkey-patching the `SampleArray` class?